unitTests/ui/SimpleFilterTest.ts

```diff
@@ -1,227 +1,327 @@
 import * as Mock from '../MockEnvironment';
```

Comment:
There's a lot of changes in that file, would you mind explaining the reasoning/logic/structure you are trying to achieve ? 

It's otherwise a bit hard to follow all the changes.

Reply:
Besides adding specific tests for the clear button behaviour, I wanted to make sure that all the previous existing tests were ok even if enableClearButton was true/false/null. Something like a 'Parameterized test' (https://medium.com/@nyablk97/parameterized-tests-with-jasmine-ecadb2856980) to avoid DRY. In this context, the parameters I used were just an array of booleans, but they could be an array of configuration options with expected results.

But maybe an overkill for the moment.

Reply:
I've committed it with the original structure and just the additional tests. Here it's easier to compare the changes https://github.com/coveo/search-ui/compare/JSUI-2686.